Develop

tarun-menta opened the pull request:
Cosmetic and styling changes, as well as a new timetable component that is more mobile friendly.

butter-chicken27 committed:
Mess card for home page added

butter-chicken27 committed:
Bus home page functionality added

tarun-menta committed:
Merge remote-tracking branch 'upstream/develop' into develop

tarun-menta committed:
Modify TimeTable to include custom events

Can now add custom events via a popup, and will be displayed in the
timetable, and saved in localStorage

tarun-menta committed:
BottomNav Bug Fix

BottomNav would highlight the default tab(home) if a page was accessed
directly via URL, instead of via the bottomNav. Fixed with using
window.location.pathname

tarun-menta committed:
Fixed lag in the timetable add event popup

Replaced react state variables with normal js variables

butter-chicken27 committed:
Fixed navbar home bug and start end time parity

butter-chicken27 committed:
Made styling changes to home and mess

butter-chicken27 committed:
Added agenda card to the home page

tarun-menta committed:
Modified TimeTable to suit mobile screens better

Changed to FullCalendar from react-big-calendar, modified css to make
the calendar fit better on a mobile screen

butter-chicken27 committed:
Modified date picker for custom events

butter-chicken27 committed:
Removed Cab Page

butter-chicken27 committed:
Added a space

tarun-menta committed:
Added theme toggle

Moved themes to separate file, added a toggle in the side navbar
Need to setup fonts and better colors, and possibly custom breakpoints

butter-chicken27 committed:
Cosmetic changes and minor bug fixes

butter-chicken27 committed:
Cosmetic changes and minor bug fixes

butter-chicken27 committed:
Cosmetic changes and minor bug fixes

butter-chicken27 committed:
Cosmetic changes and minor bug fixes

butter-chicken27 committed:
Reweighted headings

tarun-menta committed:
Saved user theme preference to localStorage

Theme was defaulting to dark on refresh, needed to store preferred theme
in localStorage

tarun-menta committed:
Updated theming on TimeTable page

Colors needed to be set with css variables to match the color theme
provided by mui. Required some tinkering with the classes provided by
FullCalendar

butter-chicken27 committed:
Added warning for timetable not synced

tarun-menta committed:
Added handling for slow or incomplete API call

Added a loading component for incomplete API call, and an error message
for API call error

butter-chicken27 committed:
Tweaked home layout and set up default expansion in mess accordion

tarun-menta committed:
Cosmetic updates to Home

Added padding to the individual cards
Times were not being updated from 'loading' until you changed pages,
issue with the useEffect dependencies, fixed.

tarun-menta committed:
Added day view to TimeTable

tarun-menta committed:
Changed timetable error message to snackbar

Error message for when aims timetable was not synced was just a header,
changed to a floating snackbar

tarun-menta committed:
Retain mess preference in localstorage

tarun-menta committed:
Updated manifest.json and added service worker

tarun-menta committed:
Changed firebase login to redirect

tarun-menta committed:
Updated firebase imports for production

Fixes the firebase warning that shows up in the console

tarun-menta committed:
Added gh-pages deployment scripts

tarun-menta committed:
Add circular loader while auth is ocurring

Previously, app would flash the sign-in page briefly before continuing
to homepage, changed to a circular loader

tarun-menta committed:
Fixed infinite loop issue

Adding date as a dependency in useEffect in Home.js caused an infinite
loop, removed

tarun-menta committed:
Added basename to router

Router needs basename for deployment on gh-pages sub-directory

butter-chicken27 committed:
Fixed home icon bug

tarun-menta committed:
Changed homepage in package.json

tarun-menta committed:
Fixed dependency issue in Home.js

tarun-menta committed:
Updated Title and status bar in index.html

Needed an extra meta tag for translucent status bar for ios pwa

butter-chicken27 committed:
Added internal app links from homepage

butter-chicken27 committed:
Merge branch 'develop' of https://github.com/peppermenta/iith-dashboard-pwa into develop

butter-chicken27 committed:
Added conditional rendering for labels based on screen size

butter-chicken27 committed:
Made changes from toggle to drop down selections and add event indentations

RachitKeertiDas committed:
Separate out Login into own component

Use a component for Login with proper styling and logo

RachitKeertiDas committed:
meta: Use correct logo in public files.

This logo is seen on the PWA in the launch screen

RachitKeertiDas committed:
Make dark theme darker

butter-chicken27 committed:
Rectified spacing to make mess menu compact

RachitKeertiDas commented:
Hi,

Can we get a review on this PR please?
If the changes are humongous to review, I can split the commits into separate Pull requests if that's preferred.

I feel that periodically merging will help in bug-spotting/fixing easily compared to merging everything in one go.

Thanks

tarun-menta committed:
Update segment start and end dates

Manual update to segment start and end dates for the new semester,
should move this to an API call in the future
